server/handlers.go

```diff
@@ -668,6 +668,44 @@ func (s *Server) handleAuthCode(w http.ResponseWriter, r *http.Request, client s
 			s.tokenErrHelper(w, errServerError, "", http.StatusInternalServerError)
 			return
 		}
+
+		tokenRef := storage.RefreshTokenRef{
+			ID:        refresh.ID,
+			CreatedAt: refresh.CreatedAt,
+			LastUsed:  refresh.LastUsed,
+		}
+
+		if session, err := s.storage.GetOfflineSession(refresh.Claims.UserID, refresh.ConnectorID); err != nil {
+			offlineSession := storage.OfflineSession{
+				UserID:  refresh.Claims.UserID,
+				ConnID:  refresh.ConnectorID,
+				Refresh: tokenRef,
+			}
+			if err := s.storage.CreateOfflineSession(offlineSession); err != nil {
+				s.logger.Errorf("failed to create offline session: %v", err)
+				s.tokenErrHelper(w, errServerError, "", http.StatusInternalServerError)
+				return
+			}
+		} else {
+			oldTokenRef := session.Refresh
```

Comment:
Per comment below, while a client can only have one refresh token per user, a user might be logged into multiple clients.